### PR TITLE
Add cronjob for removing unused imagestream tags

### PR DESCRIPTION
**Pull request by msimacek**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -48,4 +50,19 @@ osbs_pruner_successful_jobs: 5
 osbs_pruner_failed_jobs: 5
 osbs_pruner_build_clusterrole_read: system:openshift:controller:build-config-change-controller
 osbs_pruner_build_role_delete: edit
-osbs_pruner_cronjob_api_version: batch/v2alpha1
+
+# TODO we should make a separate image for cleanup jobs and not use jenkins slave
+osbs_remove_unused_istags_image: docker-registry.upshift.redhat.com/osbs/jenkins-slave-osbs-released-prod:latest
```
Comment (twaugh) [inline defaults/main.yml]: Oh, and this isn't publicly available either. :-(

File: defaults/main.yml [hunk 2]
```diff
@@ -48,4 +50,13 @@ osbs_pruner_successful_jobs: 5
...
 osbs_pruner_build_clusterrole_read: system:openshift:controller:build-config-change-controller
 osbs_pruner_build_role_delete: edit
-osbs_pruner_cronjob_api_version: batch/v2alpha1
+
+# You need to provide an image that contains oc and jq commands
+osbs_remove_unused_istags_image: ''
+osbs_remove_unused_istags_command:
+- /bin/bash
+- -c
+- "{{ lookup('template', 'remove-unused-istags.sh.j2') }}"
```
Comment (twaugh) [inline defaults/main.yml]: Oh, neat!

File: templates/remove-unused-istags.sh.j2 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    $OC get imagestreamtags -o json | \
+    jq '.items[].metadata | select(.creationTimestamp < "'${delete_threshold}'") | .name' -r | \
+    sort -u
+)
+
+set +e
+unwanted_imagestream_tags=$(
+    comm -23 <(echo "$old_imagestream_tags") <(echo "$autorebuild_imagestream_tags") | \
+    grep -v '^osbs-rcm-buildroot:' | \
+    grep -v '^osbs-test-autorebuild-from-scratch-parent:'
```
Comment (twaugh) [inline templates/remove-unused-istags.sh.j2]: Fine for now. In future we could think about making this configurable.

**Comment by athos-ribeiro** [comment]
LGTM

- The time window on `templates/remove-unused-istags.sh.j2` could also be configurable in the future.
- Is there any OSBS ticket related to this PR? If so, we usually add a bullet in the end of the commit messages pointing to it.
- Any reason for not squashing the latest 2 commits so the URLs pointing to to non-public URLs will not be in the change history?

None of these poitns should be blockers here, so address them as you will!

Thanks!!!!

**Comment by msimacek** [comment]
Adressed Athos's points except the first one - this change is about moving the script, I'd like to avoid changing more than necessary. I prefer to do one thing at a time. We'll most likely have another story for improving the job.

**Pull request merged**